# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: xpfe/appshell/AppWindow.cpp
# Commit: bf93dcd0ace3
# Full Hash: bf93dcd0ace3d0dc7253e2ed7bed688c9dedb2b8
# Author: Stephen A Pohl <spohl.mozilla.bugs@gmail.com>
# Date: 2023-12-22 21:39:32
# Description:
#   Bug 1871527: Fix a crash in headless mode due to the creation of native menu bars for modals. r=mac-reviewers,bradwerth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D197197
# ==============================================================================

diff -r b3e9fcc4a96c -r bf93dcd0ace3 xpfe/appshell/AppWindow.cpp
--- a/xpfe/appshell/AppWindow.cpp	Fri Dec 22 21:11:11 2023 +0200
+++ b/xpfe/appshell/AppWindow.cpp	Fri Dec 22 18:51:10 2023 +0000
@@ -490,11 +490,13 @@
   nsCOMPtr<nsIAppWindow> tempRef = this;
 
 #ifdef USE_NATIVE_MENUS
-  // macOS only: For modals created early in startup.
-  // (e.g. ProfileManager/ProfileDowngrade) this creates a fallback menu for
-  // the menu bar which only contains a "Quit" menu item.
-  // This allows the user to quit the application in a regular way with cmd+Q.
-  widget::NativeMenuSupport::CreateNativeMenuBar(mWindow, nullptr);
+  if (!gfxPlatform::IsHeadless()) {
+    // macOS only: For modals created early in startup.
+    // (e.g. ProfileManager/ProfileDowngrade) this creates a fallback menu for
+    // the menu bar which only contains a "Quit" menu item.
+    // This allows the user to quit the application in a regular way with cmd+Q.
+    widget::NativeMenuSupport::CreateNativeMenuBar(mWindow, nullptr);
+  }
 #endif
 
   window->SetModal(true);
